弹幕姬

AgxCOy opened the pull request:
简而言之，移植`blivechat`的布局，并适配本项目的后端。

~~目前显然还没有完工。咕咕咕。~~
感谢岛风做的收尾喵。

![wbzd](https://github.com/user-attachments/assets/d2f136e9-9a34-4813-8b7e-504a2aa15260)

AgxCOy committed:
blivechat danmaku implements (part 1)

Co-authored-by: 舰队的偶像-岛风酱! <frg2089@outlook.com>
Signed-off-by: SilverAg.L <caclx@outlook.com>

AgxCOy committed:
TODO: reimplementing blivechat danmaku component

- lazy to continue. im very vegetable.

Co-authored-by: 舰队的偶像-岛风酱! <frg2089@outlook.com>
Signed-off-by: SilverAg.L <caclx@outlook.com>

frg2089 commented on web/obs/src/components/chat/AuthorBadge.vue:
```diff
@@ -0,0 +1,54 @@
+                class="style-scope yt-icon"
+                d="M9.64589146,7.05569719 C9.83346524,6.562372 9.93617022,6.02722257 9.93617022,5.46808511 C9.93617022,3.00042984 7.93574038,1 5.46808511,1 C4.90894765,1 4.37379823,1.10270499 3.88047304,1.29027875 L6.95744681,4.36725249 L4.36725255,6.95744681 L1.29027875,3.88047305 C1.10270498,4.37379824 1,4.90894766 1,5.46808511 C1,7.93574038 3.00042984,9.93617022 5.46808511,9.93617022 C6.027222 ...
+            </g>
+          </svg>
```
应把它放到单独的 .svg 文件里
```suggestion
          <img src="xxx.svg" />
```

AgxCOy replied:
似乎没法当作标准svg看待（

frg2089 replied:
那放到单独的 .vue 文件里也是可以的 叫 PascalCaseIcon.vue

frg2089 commented on web/obs/src/components/chat/AuthorBadge.vue:
```diff
@@ -0,0 +1,54 @@
+import * as types from '../../types/ChatMessageType'
+
+const props = defineProps<{
+  isAdmin: boolean
```
改用 nullable 来写布尔值
```suggestion
  isAdmin?: boolean
```

frg2089 commented on web/obs/src/components/chat/AuthorChip.vue:
```diff
@@ -0,0 +1,52 @@
+      <AuthorBadge
+        v-if="isInMemberMessage"
+        class="style-scope yt-live-chat-author-chip"
+        :isAdmin="false"
```
布尔值改成 nullable 之后可以删掉这段
```suggestion
```

frg2089 commented on web/obs/src/components/chat/AuthorChip.vue:
```diff
@@ -0,0 +1,52 @@
+        <AuthorBadge
+          v-if="privilegeType > 0"
+          class="style-scope yt-live-chat-author-chip"
+          :isAdmin="false"
```
同 nullable
```suggestion
```

frg2089 commented on web/obs/src/components/chat/AuthorChip.vue:
```diff
@@ -0,0 +1,52 @@
+        v-if="isInMemberMessage"
+        class="style-scope yt-live-chat-author-chip"
+        :isAdmin="false"
+        :privilegeType="privilegeType" />
```
我更偏向于使用 kebab-case 来写组件的属性名
```suggestion
        :privilege-type="privilegeType" />
```

frg2089 commented on web/obs/src/components/chat/AuthorChip.vue:
```diff
@@ -0,0 +1,52 @@
+          v-if="authorType === types.AuthorType.ADMIN"
+          class="style-scope yt-live-chat-author-chip"
+          isAdmin
+          :privilegeType="0" />
```
同 kebak-case
```suggestion
          is-admin
          :privilege-type="0" />
```

frg2089 commented on web/obs/src/components/chat/AuthorChip.vue:
```diff
@@ -0,0 +1,52 @@
+          v-if="privilegeType > 0"
+          class="style-scope yt-live-chat-author-chip"
+          :isAdmin="false"
+          :privilegeType="privilegeType" />
```
同 kebab-case
```suggestion
          :privilege-type="privilegeType" />
```

frg2089 commented on web/obs/src/components/chat/AuthorChip.vue:
```diff
@@ -0,0 +1,52 @@
+import AuthorBadge from './AuthorBadge.vue'
+
+const props = defineProps<{
+  isInMemberMessage: boolean
```
同 nullable
```suggestion
  isInMemberMessage?: boolean
```

frg2089 commented on web/obs/src/components/chat/ChatItemBox.vue:
```diff
@@ -0,0 +1,63 @@
+    :time="message.time"
+    :avatarUrl="message.avatarUrl"
+    :authorName="getShowAuthorName(message)"
+    :privilegeType="message.privilegeType"
```
```suggestion
    :avatar-url="message.avatarUrl"
    :author-name="getShowAuthorName(message)"
    :privilege-type="message.privilegeType"
```

frg2089 commented on web/obs/src/components/chat/ChatItemBox.vue:
```diff
@@ -0,0 +1,63 @@
+    :avatarUrl="message.avatarUrl"
+    :authorName="getShowAuthorName(message)"
+    :price="message.price"
+    :priceText="''"
```
```suggestion
    :avatar-url="message.avatarUrl"
    :author-name="getShowAuthorName(message)"
    :price="message.price"
```

frg2089 commented on web/obs/src/components/chat/ChatItemBox.vue:
```diff
@@ -0,0 +1,63 @@
+    :authorName="message.authorName"
+    :authorType="message.authorType"
+    :privilegeType="message.privilegeType"
+    :contentParts="getShowContentParts(message)" />
```
```suggestion
    :avatar-url="message.avatarUrl"
    :author-name="message.authorName"
    :author-type="message.authorType"
    :privilege-type="message.privilegeType"
    :content-parts="getShowContentParts(message)" />
```

frg2089 commented on web/obs/src/components/chat/ChatItemBox.vue:
```diff
@@ -0,0 +1,63 @@
+    :avatarUrl="message.avatarUrl"
+    :authorName="getShowAuthorName(message)"
+    :price="message.price"
+    :priceText="message.price <= 0 ? getGiftShowNameAndNum(message) : ''"
```
```suggestion
    :avatar-url="message.avatarUrl"
    :author-name="getShowAuthorName(message)"
    :price="message.price"
    :price-text="message.price <= 0 ? getGiftShowNameAndNum(message) : ''"
```

frg2089 commented on web/obs/src/components/chat/ChatItemBox.vue:
```diff
@@ -0,0 +1,63 @@
+} from './constants'
+import MembershipItem from './MembershipItem.vue'
+import PaidMessage from './PaidMessage.vue'
+import TextMessage from './TextMessage.vue'
```
组件应自动导入 而不是写在这里
```suggestion
```

AgxCOy replied:
浇浇（

frg2089 commented on web/obs/src/components/chat/ChatItemBox.vue:
```diff
@@ -0,0 +1,63 @@
+  {
+    showGiftName: false,
+  },
+)
```
这里没必要用 withDefaults
null undefined 都可以当 false 用
```suggestion
defineProps<{
  message: types.AnyDisplayMessage
  showGiftName?: boolean
}>()
```

AgxCOy replied:
L20 附近的`getGiftShowContent(message, showGiftName)`不许传 null。不然像 c# 一样 fallback：`showGiftName ?? false`？

frg2089 commented on web/obs/src/components/chat/constants.ts:
```diff
@@ -0,0 +1,198 @@
+ */
+export function getShowGuardLevelText(guardLevel: GuardLevel) {
+  return GuardLevel.getText(guardLevel)
+}
```
箭头函数 yes! (
```suggestion
export const getShowGuardLevelText = (guardLevel: GuardLevel) => GuardLevel.getText(guardLevel)
```

frg2089 commented on web/obs/src/components/chat/constants.ts:
```diff
@@ -0,0 +1,198 @@
+  if (!showGiftName) return ''
+
+  return `赠送 ${getGiftShowNameAndNum(message)}`
+}
```
```suggestion
export const getPriceConfig = (price: number = 0) => {
  let i = 0
  // 根据先验知识，从小找到大通常更快结束
  for (; i < PRICE_CONFIGS.length - 1; i++) {
    let nextConfig = PRICE_CONFIGS[i + 1]
    if (price < nextConfig.price) {
      return PRICE_CONFIGS[i]
    }
  }
  return PRICE_CONFIGS[i]
}

export const getShowContent = (message: AnyDisplayMessage): string | undefined => {
  if ('translation' in message)
    return `${message.content}（${message.translation}）`

  return (message as any)?.content
}

export const getShowContentParts = (message: TextMessage) => {
  let contentParts = [...message.contentParts]
  if (message.translation) {
    contentParts.push({
      type: ContentPartType.TEXT,
      text: `（${message.translation}）`,
    })
  }
  return contentParts
}

export const getGiftShowContent = (
  message: GiftMessage,
  showGiftName: boolean,
) => {
  if (!showGiftName) return ''

  return `赠送 ${getGiftShowNameAndNum(message)}`
}
```

frg2089 commented on web/obs/src/components/chat/index.vue:
```diff
@@ -0,0 +1,105 @@
+    <Ticker
+      class="style-scope yt-live-chat-renderer"
+      v-model:messages="paidMessages"
+      :showGiftName="showGiftName" />
```
```suggestion
      :show-gift-name="showGiftName" />
```

frg2089 commented on web/obs/src/components/chat/ImgShadow.vue:
```diff
@@ -0,0 +1,42 @@
+  () => props.imgUrl,
+  v => {
+    showImgUrl.value = v
+  },
```
```suggestion
  v => showImgUrl.value = v,
```

AgxCOy replied:
不怪我喵 prettier干的喵（

frg2089 commented on web/obs/src/components/chat/ImgShadow.vue:
```diff
@@ -0,0 +1,42 @@
+const onLoadError = () => {
+  if (showImgUrl.value !== constants.DEFAULT_AVATAR_URL) {
+    showImgUrl.value = constants.DEFAULT_AVATAR_URL
+  }
```
应提前 return 避免不必要的嵌套
```suggestion
  if (showImgUrl.value === constants.DEFAULT_AVATAR_URL) return
  
  showImgUrl.value = constants.DEFAULT_AVATAR_URL
```

frg2089 commented on web/obs/src/components/chat/index.vue:
```diff
@@ -0,0 +1,105 @@
+            <ChatItemBox
+              v-for="message in messages"
+              :key="message.id"
+              :showGiftName="props.showGiftName"
```
```suggestion
              :show-gift-name="props.showGiftName"
```

frg2089 commented on web/obs/src/components/chat/index.vue:
```diff
@@ -0,0 +1,105 @@
+    maxNumber: 60,
+    showGiftName: false,
+  },
+)
```
```suggestion
const props = withDefaults(
  defineProps<{
    maxNumber: number
    showGiftName?: boolean
  }>(),
  {
    maxNumber: 60,
  },
)
```

frg2089 commented on web/obs/src/components/chat/MembershipItem.vue:
```diff
@@ -0,0 +1,66 @@
+          height="40"
+          width="40"
+          class="style-scope yt-live-chat-membership-item-renderer"
+          :imgUrl="avatarUrl" />
```
```suggestion
          :img-url="avatarUrl" />
```

frg2089 commented on web/obs/src/components/chat/MembershipItem.vue:
```diff
@@ -0,0 +1,66 @@
+                isInMemberMessage
+                :authorName="authorName"
+                :authorType="0"
+                :privilegeType="privilegeType" />
```
```suggestion
                is-in-member-message
                :author-name="authorName"
                :author-type="0"
                :privilege-type="privilegeType" />
```

frg2089 commented on web/obs/src/components/chat/MembershipItem.vue:
```diff
@@ -0,0 +1,66 @@
+// import * as utils from '@/utils'
+import * as types from '../../types/ChatMessageType'
+import AuthorChip from './AuthorChip.vue'
+import ImgShadow from './ImgShadow.vue'
```
应自动导入
```suggestion
```

frg2089 commented on web/obs/src/components/chat/index.vue:
```diff
@@ -0,0 +1,105 @@
+import ChatItemBox from './ChatItemBox.vue'
+import * as constants from './constants'
+import { DanmakuQueue } from './DanmakuQueue'
+import Ticker from './Ticker.vue'
```
自动导入
```suggestion
import * as types from '../../types/ChatMessageType'
import * as constants from './constants'
import { DanmakuQueue } from './DanmakuQueue'
```

frg2089 commented on web/obs/src/components/chat/index.vue:
```diff
@@ -0,0 +1,105 @@
+const refreshCantScrollStartTime = () => {
+  if (cantScrollStartTime.value) {
+    cantScrollStartTime.value = new Date()
+  }
```
```suggestion
  if (!cantScrollStartTime.value) return

  cantScrollStartTime.value = new Date()
```

frg2089 commented on web/obs/src/components/chat/PaidMessage.vue:
```diff
@@ -0,0 +1,82 @@
+          height="40"
+          width="40"
+          class="style-scope yt-live-chat-paid-message-renderer"
+          :imgUrl="avatarUrl" />
```
```suggestion
          :img-url="avatarUrl" />
```

frg2089 commented on web/obs/src/components/chat/PaidMessage.vue:
```diff
@@ -0,0 +1,82 @@
+
+<script lang="ts" setup>
+import * as constants from './constants'
+import ImgShadow from './ImgShadow.vue'
```
```suggestion
```

frg2089 commented on web/obs/src/components/chat/TextMessage.vue:
```diff
@@ -0,0 +1,65 @@
+      height="24"
+      width="24"
+      class="style-scope yt-live-chat-text-message-renderer"
+      :imgUrl="avatarUrl" />
```
```suggestion
      :img-url="avatarUrl" />
```

frg2089 commented on web/obs/src/components/chat/TextMessage.vue:
```diff
@@ -0,0 +1,65 @@
+        :isInMemberMessage="false"
+        :authorName="authorName"
+        :authorType="authorType"
+        :privilegeType="privilegeType" />
```
```suggestion
        :is-in-member-message="false"
        :author-name="authorName"
        :author-type="authorType"
        :privilege-type="privilegeType" />
```

frg2089 commented on web/obs/src/components/chat/TextMessage.vue:
```diff
@@ -0,0 +1,65 @@
+
+import * as types from '../../types/ChatMessageType'
+import AuthorChip from './AuthorChip.vue'
+import ImgShadow from './ImgShadow.vue'
```
```suggestion
```

frg2089 commented on web/obs/src/components/chat/Ticker.vue:
```diff
@@ -0,0 +1,273 @@
+const emit = defineEmits<{
+  (
+    e: 'update:messages',
+    value: Exclude<types.AnyDisplayMessage, types.TextMessage>[],
```
就 还是喜欢直接叫它 ev
```suggestion
    ev: Exclude<types.AnyDisplayMessage, types.TextMessage>[],
```

AgxCOy replied:
（

frg2089 commented on web/obs/src/components/chat/Ticker.vue:
```diff
@@ -0,0 +1,273 @@
+                height="24"
+                width="24"
+                class="style-scope yt-live-chat-ticker-paid-message-item-renderer"
+                :imgUrl="message.raw.avatarUrl" />
```
```suggestion
                :img-url="message.raw.avatarUrl" />
```

frg2089 commented on web/obs/src/components/chat/Ticker.vue:
```diff
@@ -0,0 +1,273 @@
+        class="style-scope yt-live-chat-ticker-renderer"
+        :avatarUrl="pinnedMessage.avatarUrl"
+        :authorName="getShowAuthorName(pinnedMessage)"
+        :privilegeType="pinnedMessage.privilegeType"
```
```suggestion
        :avatar-url="pinnedMessage.avatarUrl"
        :author-name="getShowAuthorName(pinnedMessage)"
        :privilege-type="pinnedMessage.privilegeType"
```

frg2089 commented on web/obs/src/components/chat/Ticker.vue:
```diff
@@ -0,0 +1,273 @@
+        v-else
+        class="style-scope yt-live-chat-ticker-renderer"
+        :price="pinnedMessage.price"
+        :priceText="''"
```
```suggestion
```

frg2089 commented on web/obs/src/components/chat/Ticker.vue:
```diff
@@ -0,0 +1,273 @@
+        :price="pinnedMessage.price"
+        :priceText="''"
+        :avatarUrl="pinnedMessage.avatarUrl"
+        :authorName="getShowAuthorName(pinnedMessage)"
```
```suggestion
        :avatar-url="pinnedMessage.avatarUrl"
        :author-name="getShowAuthorName(pinnedMessage)"
```

frg2089 commented on web/obs/src/components/chat/Ticker.vue:
```diff
@@ -0,0 +1,273 @@
+  ): void
+}>()
+
+const showMessages = computed(() => {
+  let res = []
+  for (let message of props.messages) {
+    if (!needToShow(message)) {
+      continue
+    }
+    res.push({
+      raw: message,
+      bgColor: getBgColor(message),
+      color: getColor(message),
+      text: getText(message),
+    })
+  }
+  return res
+})
```
```suggestion
const showMessages = computed(() => props.messages
  .filter(needToShow)
  .map(i => ({
    raw: i,
    bgColor: getBgColor(i),
    color: getColor(i),
    text: getText(i),
  })))
```

AgxCOy replied:
不懂 linq 导致的（

frg2089 commented on web/obs/src/components/chat/Ticker.vue:
```diff
@@ -0,0 +1,273 @@
+const pinnedMessageShowContent = computed(() => {
+  if (!pinnedMessage.value) {
+    return ''
+  }
```
```suggestion
  if (!pinnedMessage.value) return ''

```

frg2089 commented on web/obs/src/components/chat/Ticker.vue:
```diff
@@ -0,0 +1,273 @@
+    return getGiftShowContent(pinnedMessage.value, props.showGiftName)
+  } else {
+    return getShowContent(pinnedMessage.value)
+  }
```
```suggestion
  if (pinnedMessage.value.type === types.MessageType.GIFT)
    return getGiftShowContent(pinnedMessage.value, props.showGiftName)

  return getShowContent(pinnedMessage.value)
```

frg2089 commented on web/obs/src/components/chat/Ticker.vue:
```diff
@@ -0,0 +1,273 @@
+
+const curTime = ref(new Date())
+const updateTimerId = ref(window.setInterval(updateProgress, 1000))
+const pinnedMessage = ref<Exclude<types.AnyDisplayMessage, types.TextMessage>>() // really?
```
这东西怎么在最后？

要么放在用到它的方法前 要么放开头

AgxCOy replied:
`updateTimerId`需要扔这（它调的`updateProgress`方法）

frg2089 commented on web/obs/src/components/chat/ChatMessageType.ts:
这个文件不应该放在这里 可以和那些组件放在一起

frg2089 commented on web/obs/src/types/ChatMessageType.ts:
```diff
@@ -0,0 +1,200 @@
+/** 一段内容 */
+export type AnyContentPart = TextContentPart | ImageContentPart
+
+/** */
```
```suggestion
/** 一段文本内容 */
```

frg2089 commented on web/obs/src/types/ChatMessageType.ts:
```diff
@@ -0,0 +1,200 @@
+}
+
+/** 醒目留言消息 */
+
```
```suggestion
```

AgxCOy replied:
这注释不留着马（

frg2089 replied:
我这里是删除第 188 行 那是个空白行
JsDoc 最好是紧贴着写

frg2089 commented on web/obs/src/components/chat/ChatMessageType.ts:
```diff
@@ -0,0 +1,200 @@
+  /** 礼物图标URL */
+  giftIconUrl: string
+}
+/** 上舰消息 */
```
```suggestion

/** 上舰消息 */
```

frg2089 commented on pnpm-lock.yaml:
```diff
@@ -2253,6 +2242,8 @@ snapshots:
   date-fns@3.6.0: {}
 
+  date-fns@4.1.0: {}
+
```
emmm...

AgxCOy replied:
依赖没升导致的。已修。

frg2089 commented on web/obs/src/components/chat/DanmakuQueue.ts:
这个文件不应该在这里 挪到其他地方更好些

frg2089 reviewed (changes requested):
先进行一个代码的审

基本都是些小的格式问题

以及 那个 date-fns 3.6 不知道是哪里来的 最好处理掉

frg2089 reviewed (changes requested):
先进行一个代码的审

基本都是些小的格式问题

以及 那个 date-fns 3.6 不知道是哪里来的 最好处理掉

AgxCOy commented:
整改（change as requested）完之后还是考虑先读懂岛风demo写的是怎么回事= =现在是莫得厮路。

AgxCOy committed:
拟接受大部分修改意见

- `DanmakuQueue.ts`不知放哪合适，暂不动
- 暂定将svg分离进`ModeratorIcon.vue`.

Signed-off-by: SilverAg.L <caclx@outlook.com>

AgxCOy committed:
rename `ModeratorIcon.vue` into static svg.

Signed-off-by: SilverAg.L <caclx@outlook.com>

AgxCOy committed:
适配 blivechat 布局&处理流

TODO:
- 在 3115c4bce 基础上重新实现布局所需 API（即 original ... logic 注释块）
- `EventClient.ts`标注的其余待办

Signed-off-by: SilverAg.L <caclx@outlook.com>

frg2089 commented on web/obs/src/components/chat/AuthorChip.vue:
```diff
@@ -0,0 +1,50 @@
+        class="style-scope yt-live-chat-author-chip"
+        :privilege-type="privilegeType" />
+      <template v-else>
+        <AuthorBadge
+          v-if="authorType === types.AuthorType.ADMIN"
+          class="style-scope yt-live-chat-author-chip"
+          is-admin
+          :privilege-type="0" />
+        <AuthorBadge
+          v-if="privilegeType > 0"
+          class="style-scope yt-live-chat-author-chip"
+          :privilege-type="privilegeType" />
```
这里应该可以想办法合并一下

```suggestion
        <AuthorBadge
          class="style-scope yt-live-chat-author-chip"
          :is-admin="authorType === types.AuthorType.ADMIN"
          :privilege-type="privilegeType" />
```

frg2089 commented on web/obs/src/components/chat/AuthorChip.vue:
```diff
@@ -0,0 +1,50 @@
+</template>
+
+<script lang="ts" setup>
+import AuthorBadge from './AuthorBadge.vue'
```
如无意外 应该由 AutoImport 插件自动处理 .vue 组件的导入

frg2089 commented on web/obs/src/components/chat/EventClient.ts:
```diff
@@ -0,0 +1,150 @@
+  viewed = ref(0)
+
+  /** 外显消息队列 */
+  messages = reactive<types.AnyDisplayMessage[]>([])
```
仔细想想 或许这里不该写 vue 响应式对象 改成用事件往外发更好些
响应式对象就让它在SFC里和pinia里好了

https://developer.mozilla.org/docs/Web/API/EventTarget

AgxCOy replied:
那是要继承它马（
那话又说回来，`message[]`要在`[roomId].vue`里搞马，就像 blivechat 一样？

frg2089 commented on web/obs/src/pages/chat/[roomId].vue:
```diff
@@ -1,215 +1,139 @@
+  {
+    maxNumber: 60,
+  },
+)
```
这个组件是在 pages 目录下的 可能预期目的是作为一个页组件存在的
页组件被 RouterView 动态渲染，它通常不会传递属性 这里始终会保持默认

如果需要获取参数，应该从 `ReturnType<typeof useRoute>` 对象里取它的 Query 参数

或者你要实在想用 defineProps 也行 研究一下这两个东西

https://router.vuejs.org/guide/essentials/passing-props
https://uvr.esm.is/guide/extending-routes#definepage

frg2089 commented on web/obs/src/pages/chat/[roomId].vue:
```diff
@@ -1,215 +1,139 @@
-const hot = ref(0)
-const data = reactive<Bili.Live.Danmaku.Event[]>([])
 const user = useUser()
+await user.updateUserInfo()
```
这里似乎不在某个方法里 如果需要在组件可用时调用异步方法 把调用的部分放到 OnMounted 里

frg2089 commented on web/obs/src/stores/user.ts:
```diff
@@ -15,6 +16,15 @@ export const useUser = defineStore('user', () => {
+        },
+      },
+    })
+    roomId.value = resRoom.data?.roomid ?? undefined
```
你猜怎么着？我有非空断言洁癖 我看到 ! 就不自在

```suggestion
    if (!userId.value) return
    const resRoom = await client.GET('/bili/live/infoByUid', {
      params: {
        query: {
          userId: userId.value,
        },
      },
    })
    roomId.value = resRoom.data?.roomid ?? undefined
```

frg2089 reviewed (changes requested):
对提交进行一个快速审查

frg2089 committed:
refactor(event): 重构 SSE 解析为 parse-sse

Signed-off-by: 舰队的偶像-岛风酱! <frg2089@outlook.com>

frg2089 committed:
fix: 优化弹幕客户端资源管理和布局样式

Signed-off-by: 舰队的偶像-岛风酱! <frg2089@outlook.com>

frg2089 committed:
feat: add custom CSS support and improve chat layout

Signed-off-by: 舰队的偶像-岛风酱! <frg2089@outlook.com>

frg2089 committed:
refactor: rename chat component and update import paths

Signed-off-by: 舰队的偶像-岛风酱! <frg2089@outlook.com>

frg2089 committed:
feat: add chat message sender component

Signed-off-by: 舰队的偶像-岛风酱! <frg2089@outlook.com>

frg2089 committed:
refactor: remove redundant roomId logic from user store

Signed-off-by: 舰队的偶像-岛风酱! <frg2089@outlook.com>
